extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientRecorder.java

```diff
@@ -142,15 +142,17 @@ public OidcClient apply(String tokenRequestUri, Throwable t) {
                                 : OidcConstants.PASSWORD_GRANT;
                         setGrantClientParams(oidcConfig, tokenGrantParams, grantType);
 
-                        Map<String, String> grantOptions = oidcConfig.getGrantOptions()
-                                .get(oidcConfig.grant.getType().name().toLowerCase());
-                        if (oidcConfig.grant.getType() == Grant.Type.PASSWORD) {
-                            tokenGrantParams.add(OidcConstants.PASSWORD_GRANT_USERNAME,
-                                    grantOptions.get(OidcConstants.PASSWORD_GRANT_USERNAME));
-                            tokenGrantParams.add(OidcConstants.PASSWORD_GRANT_PASSWORD,
-                                    grantOptions.get(OidcConstants.PASSWORD_GRANT_PASSWORD));
-                        } else if (grantOptions != null && oidcConfig.grant.getType() == Grant.Type.CLIENT) {
-                            tokenGrantParams.addAll(grantOptions);
+                        if (oidcConfig.getGrantOptions() != null) {
```

Comment:
@radcortez could you have a look at this? I wonder if it's not a side effect of one of your patches and might be undesired?

Reply:
Sure. I'll have a look.